src/elements/content-sidebar/activity-feed/comment/BaseComment.js

```diff
@@ -0,0 +1,302 @@
+
+    const [isConfirmingDelete, setIsConfirmingDelete] = React.useState<boolean>(false);
+    const [isEditing, setIsEditing] = React.useState<boolean>(false);
+    const [isInputOpen, setIsInputOpen] = React.useState<boolean>(false);
+
+    const selectComment = (isSelected: boolean = true): void => {
+        onSelect(isSelected);
+    };
+
+    const handleDeleteConfirm = (): void => {
+        onDelete({ id, permissions });
+        selectComment(false);
+    };
+
+    const handleDeleteCancel = (): void => {
+        setIsConfirmingDelete(false);
+        selectComment(false);
+    };
+
+    const handleDeleteClick = (): void => {
+        setIsConfirmingDelete(true);
+        selectComment();
+    };
+
+    const handleEditClick = (): void => {
+        setIsEditing(true);
+        setIsInputOpen(true);
+        selectComment();
+    };
+
+    const handleMenuClose = (): void => {
+        if (isConfirmingDelete || isEditing || isInputOpen) {
+            return;
+        }
+        selectComment(false);
+    };
+
+    const handleMenuOpen = (): void => {
+        selectComment();
+    };
+
+    const commentFormFocusHandler = (): void => {
+        setIsInputOpen(true);
+        selectComment();
+    };
+
+    const commentFormCancelHandler = (): void => {
+        setIsInputOpen(false);
+        setIsEditing(false);
+        selectComment(false);
+    };
+
+    const commentFormSubmitHandler = (): void => {
+        setIsInputOpen(false);
+        setIsEditing(false);
+        selectComment(false);
+    };
+
+    const handleMessageUpdate = ({
+        id: messageID,
+        text,
+        hasMention,
+    }: {
+        hasMention: boolean,
+        id: string,
+        text: string,
+    }): void => {
+        onEdit(messageID, text, undefined, hasMention, permissions);
+        commentFormSubmitHandler();
+    };
+
+    const handleStatusUpdate = (selectedStatus: FeedItemStatus): void => {
+        onEdit(id, undefined, selectedStatus, false, permissions);
+    };
+
+    const canDelete = permissions.can_delete;
+    const canEdit = onEdit !== noop && permissions.can_edit;
+    const canResolve = onEdit !== noop && permissions.can_resolve;
+    const createdAtTimestamp = new Date(created_at).getTime();
+    const createdByUser = created_by || PLACEHOLDER_USER;
+    const isEdited = modified_at !== undefined && modified_at !== created_at;
+    const isMenuVisible = (canDelete || canEdit || canResolve) && !isPending;
+    const isResolved = status === COMMENT_STATUS_RESOLVED;
+
+    return (
+        <ActivityCard className="bcs-Comment">
+            <Media
+                className={classNames('bcs-Comment-media', {
+                    'bcs-is-pending': isPending || error,
+                })}
+            >
+                <Media.Figure>
+                    <Avatar getAvatarUrl={getAvatarUrl} user={createdByUser} />
+                </Media.Figure>
+                <Media.Body>
+                    {isMenuVisible && (
```

Comment:
So this component is getting kind of big. Keep in mind that when we add the replies it will get bigger. Can we create a separate component for the menu?